### ISSUE TITLE
Question about softmax usage

### ISSUE THREAD

**Issue by tae-mo** [siyueyu/DCFM#5]
Hi, I am really interested in your work.

While studying your code, I got a question in here:
https://github.com/siyueyu/DCFM/blob/05800b67ccf70f9ed55dd1f33ee1ff3b3503eb09/models/main.py#L169

I don't really get why that line is necessary.

I would appreciate if you give me any insights!

Thank you

**Comment by siyueyu** [comment]
@tae-mo you can try to remove it.  I thought it might influence the backpropagation gradients. So I didn/t delete it.
